From a9a01812fdf49c0b0a7eb3828220f098a9a8f4dc Mon Sep 17 00:00:00 2001
Message-Id: <a9a01812fdf49c0b0a7eb3828220f098a9a8f4dc.1413836944.git.chang-joon.lee@intel.com>
In-Reply-To: <5d8aa9994fff43965a6fab8c9a528b8b47a9d624.1413836944.git.chang-joon.lee@intel.com>
References: <5d8aa9994fff43965a6fab8c9a528b8b47a9d624.1413836944.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 15 Oct 2014 08:28:49 +0530
Subject: [PATCH 67/71] FOR_UPSTREAM [VPG]: drm/i915: Program IA/GT Power BIAS

Programming the right Power BIAS between IA/GT for better rps/turbo performance.

Issue: GMINL-2350
Change-Id: Icb85f957c540dde9f219231c28aa1a0d0d75bdab
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h |    5 +++++
 drivers/gpu/drm/i915/intel_pm.c |    7 +++++++
 2 files changed, 12 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index eed6154..e7eafc4 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -611,6 +611,11 @@ enum punit_power_well {
 #define   FB_FMAX_VMIN_FREQ_LO_SHIFT		27
 #define   FB_FMAX_VMIN_FREQ_LO_MASK		0xf8000000
 
+#define VLV_IOSFB_RPS_OVERRIDE                  0x04
+#define VLV_OVERRIDE_RPS_REG                    1
+#define VLV_ENABLE_TDP_SHARE			(1 << 1)
+#define VLV_BIAS_VAL				(6 << 2)
+
 #define VLV_CZ_CLOCK_TO_MILLI_SEC		100000
 #define VLV_RP_UP_EI_THRESHOLD			90
 #define VLV_RP_DOWN_EI_THRESHOLD		70
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 4a0cc8c..cc6dd50 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -4810,6 +4810,13 @@ static void cherryview_enable_rps(struct drm_device *dev)
 			 vlv_gpu_freq(dev_priv, dev_priv->rps.efficient_freq),
 			 dev_priv->rps.efficient_freq);
 
+	/* Setting Fixed Power Bias */
+	val = VLV_OVERRIDE_RPS_REG
+		| VLV_ENABLE_TDP_SHARE
+		| VLV_BIAS_VAL;
+
+	vlv_punit_write(dev_priv, VLV_IOSFB_RPS_OVERRIDE, val);
+
 	valleyview_set_rps(dev_priv->dev, dev_priv->rps.efficient_freq);
 
 	vlv_set_rps_mode(dev, false);
-- 
1.7.9.5

